dit/dit.py

```diff
@@ -240,25 +269,28 @@ def _load_index(self):
                 self.index = json.load(f)
 
     def _rebuild_index(self):
-        self.index = [[None, [[None, []]]]]
+        self.index = [[self.root_name, [[self.root_name, []]]]]
         c_group = None
         c_subgroup = None
-        for root, dirs, files in os.walk(self._base_path()):
+        base_path = self._base_path()
+        for root, dirs, files in os.walk(base_path):
             for f in files:
                 if not self._is_valid_name(f):
                     continue
-                p = root[len(self._base_path()) + 1:].split("/")
+                p = root[len(base_path) + 1:].split("/")
                 p = [i for i in p if i]
                 if len(p) == 0:
-                    group, subgroup = None, None
+                    group, subgroup = self.root_name, self.root_name
                 elif len(p) == 1:
-                    group, subgroup = None, p[0]
-                else:
+                    group, subgroup = self.root_name, p[0]
+                elif len(p) == 2:
                     group, subgroup = p
+                else:
```

Comment:
Someone might create a directory in the wrong place... It does not hurt to check.

Trivial checks like this should be done as much as possible.